src/test/regress/sql/multi_task_assignment_policy.sql

```diff
@@ -9,6 +9,28 @@ SET citus.next_shard_id TO 880000;
 SHOW server_version \gset
 SELECT substring(:'server_version', '\d+')::int > 9 AS version_above_nine;
 
+-- the function simply parses the results and returns 'shardId@worker'
+-- for all the explain task outputs
+CREATE OR REPLACE FUNCTION parse_explain_output(in qry text, in table_name text, out r text) 
+RETURNS SETOF TEXT AS $$
+DECLARE
+       portOfTheTask text;
+       shardOfTheTask text;
+begin
+  for r in execute qry loop
+       IF r LIKE '%port%' THEN
+       portOfTheTask = substring(r, '([0-9]{1,10})');
+    END IF;
+
+    IF r LIKE '%' || table_name || '%' THEN
+       shardOfTheTask = substring(r, '([0-9]{1,10})');
+       return QUERY SELECT  shardOfTheTask || '@' || portOfTheTask ;
```

Comment:
That's some pretty nasty hacking, but I'll allow it because debug messages are bad.

(would it make sense to use JSON output? or would that make this function super verbose?)

Reply:
> That's some pretty nasty hacking, but I'll allow it because debug messages are bad.

The problem is that we cannot use DEBUG outputs because they might change -- because we're now relying on the Postgres' transactionId.

> That's some pretty nasty hacking,

I completely agree this. But, I really couldn't find anything better.

Reply:
Any suggestion to simplify it? What I try is simply make sure that `count(DISTINCT)` of `shardId@port` is 1 or 2 depending on the test.

Reply:
My first suggestion would be to use JSON output in the explain and then cast it to jsonb, but the queries might get pretty nasty and then still require some parsing.

The line parsing using regexes might break if explain output changes (or even if the table name contains a number) and it may not be immediately obvious from the regression test failure that this is the problem. It's probably helpful to at least show the actual shard IDs and ports in the output.

Reply:
(but I'm fine with merging this as is)